components/ai_chat/resources/common/useAPIState.ts

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import * as React from 'react'
+import API from './api'
+
+export default function useAPIState<T, Y>(api: API<Y>, defaultState: T) {
```

Comment:
:+1: this looks great, I am 100% on board

Reply:
Actually, can't this just be:
```
export default function useAPIState<T>(api: API<T>, defaultState: T) {
```